diesel_cli/src/database.rs

```diff
@@ -285,10 +285,27 @@ pub fn database_url(matches: &ArgMatches) -> String {
 
 #[cfg(any(feature="postgres", feature="mysql"))]
 fn change_database_of_url(database_url: &str, default_database: &str) -> (String, String) {
-    let mut split: Vec<&str> = database_url.split('/').collect();
-    let database = split.pop().unwrap();
-    let new_url = format!("{}/{}", split.join("/"), default_database);
-    (database.to_owned(), new_url)
+    // the database name will always start with the 3rd forward slash and end with
+    // the end of the string (postgres only: or the next '?').
```

Comment:
Can you expand on the intention of this splitting a bit (connection strings using sockets instead of host name by using the query param form)? I'm not sure I can follow this when I see it again in a few month. Maybe also add the "template" version of a connection string, and explicitly mention the two forms, one using the user@host/db form, and the other using query params.